src/query-run-metadata.ts

```diff
@@ -5,6 +5,7 @@ export interface QueryRunMetadata {
   nwo: string;
   resultCount?: number;
   sha?: string;
+  sourceLocationPrefix?: string;
```

Comment:
Should this be optional? In `interpret.ts` we throw an error if it isn't defined.

Reply:
Same point for `resultCount` actually. It's optional in this type but then we throw later if it isn't defined.

Also, if there's reasons and you've decided this is the pattern you want to go for that's fine.

Reply:
Good question! If the query run failed, the metadata won't contain any of `sha`/`resultCount`/`sourceLocationPrefix` (just the `nwo`) so I couldn't make them required 🤔

Reply:
Oh I see, because the `metadata` is for both successful and failed runs, but only in the success case are `resultsCount`/`sourceLocationPrefix` mandatory.

LGTM then. Thanks for explaining! 👍